### PR TITLE
collator-protocol: remove `elastic-scaling-experimental` feature

### PR DESCRIPTION
**Pull request by sandreim**
Validators already have been upgraded so they could already receive the new `CollationWithParentHeadData` response when fetching collation. However this is only sent by collators when the parachain has more than 1 core is assigned.

TODO:
- [x] PRDoc

### CODE REVIEW

**Review by lexnv — APPROVED** [review]

**Review by tdimitrov — APPROVED** [review]

File: prdoc/pr_4595.prdoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+title: "Remove `elastic-scaling-experimental` feature flag"
+
+doc:
+  - audience: Node Dev
+    description: |
+      The feature was masking the ability of collators to respond with `CollationWithParentHeadData`
+      to validator collation fetch requests. This is no longer needed as validators 
+      have been upgraded to support the new response. `CollationWithParentHeadData` however
```
Comment (alindima) [inline prdoc/pr_4595.prdoc:8]: this is not really true for all validators. I would add a section for Node Operator encouraging to upgrade and saying what will happen if they don't

Comment (sandreim) [reply prdoc/pr_4595.prdoc]: done!

**Review by alindima — APPROVED** [review]

File: prdoc/pr_4595.prdoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    description: |
+      This change enables elastic scaling support in collators. Please upgrade to latest version, 
+      otherwise validator nodes will not be able to back elastic parachain blocks leading to 
+      missed rewards. 
+
+crates:
+  - name: polkadot-collator-protocol
+    bump: minor
+  - name: polkadot-service
+    bump: minor
```
Comment (bkchr) [inline prdoc/pr_4595.prdoc:20]:
```suggestion
  - name: polkadot-collator-protocol
    bump: minor
    validate: false
  - name: polkadot-service
    bump: minor
    validate: false
```

**Review by bkchr — APPROVED** [review]

File: prdoc/pr_4595.prdoc [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      This change enables elastic scaling support in collators. Please upgrade to latest version, 
+      otherwise validator nodes will not be able to back elastic parachain blocks leading to 
+      missed rewards. 
+
+crates:
+  - name: polkadot-collator-protocol
+    bump: minor
+    validate: false
+  - name: polkadot-service
+    bump: minor
```
Comment (bkchr) [inline prdoc/pr_4595.prdoc:21]:
```suggestion
    bump: major
    validate: false
  - name: polkadot-service
    bump: major
```

**Pull request merged**